project1/applications.py

```diff
@@ -47,6 +49,38 @@ def result():
     else:
         return render_template("Registration.html")
 
+@app.route("/bookpage", methods=[ 'POST', 'GET'])
+def review_page():
+    db.create_all()
+    if request.method == 'POST':
+        r = reviews(request.form["isbn"], request.form["user"], request.form["rate"], request.form["review"])
+        data2 = reviews.query.filter_by(user = request.form['user']).first()
+        data1= reviews.query.filter_by(isbn = request.form['isbn']).first()
+        temp = list(request.form.items())
+        # print(temp)
+        Isbn = temp[0][1]
+        # details = reviews.query.all()
+        details = db.session.query(reviews).filter(reviews.isbn == Isbn)
+        if data2 is not None and data1 is not None:
+            print("Sorry, You have already reviewed  book")
+            message = "Error: Sorry, You have already reviewed  book"
+            return render_template("bookpage.html", message1  = message, details = details)
+        print(r)
+        # print(db.query.all())
+        db.session.add(r)
+        db.session.commit()
```

Comment:
Always keep db.session.add() and commit() in try and except. So if the database is crashed or under maintenance, data won't be added. Display message should be sent accordingly